src/util/convertHTMLtoJSON.ts

```diff
@@ -0,0 +1,156 @@
+import _ from 'lodash'
+import { Element, HimalayaNode, Text, parse } from 'himalaya'
+import { Block } from '../action-creators/importText'
+
+interface PreBlock extends Block {
+  partOfThought?: boolean,
+}
+
+/** Parses input HTML and saves in JSON array using Himalaya. */
+export const convertHTMLtoJSON = (html: string) => {
+
+  /** Check if string contains only characters from given 'includes' array. */
+  const isStringIncludesOnly = (str: string, includes: string[]) => {
+    const regExp = new RegExp(`^[${includes.join('')}]+$`, 'g')
+    return regExp.test(str)
+  }
+
+  /** Retrieve attribute from Element node by key. */
+  const getAttribute = (key: string, node: Element) => {
+    const { attributes } = node
+    const attribute = attributes.find(attr => attr.key === key)
+    return attribute ? attribute.value : undefined
+  }
+
+  /** Removes empty nodes and comments from himalaya's JSON output. */
+  const removeEmptyNodesAndComments = (nodes: HimalayaNode[]): (Element | Text)[] => {
+    const filteredNodes = nodes.filter(node => {
+      if (node.type === 'element') {
+        if (node.tagName === 'br') return false
+        return true
+      }
+      if (node.type === 'comment') return false
+      // remove text nodes containing only empty space and new line characters, such nodes could be created by himalaya in case of multi-line html input
+      if (node.type === 'text' && isStringIncludesOnly(node.content, ['\n', ' '])) return false
+      return node.content.length
```

Comment:
Please re-write the filter predicate as a single (nested) boolean expression. You can break the boolean onto separate lines to make it easier to read. The current use of `if` statements and explicit boolean returns does not fit the existing style of the project.

Reply:
@pushkov-fedor